firmware/watchguard_sysa_dl.ksy

```diff
@@ -0,0 +1,137 @@
+meta:
+  id: watchguard_sysa_dl
+  title: Watchguard Fireware web update (Sysa-dl) file format
+  file-extension: sysa-dl
+  license: MIT
+  endian: be
+
+seq:
+  - id: fw_header
+    size: 24
+    type: fw_header
+
+  - id: sections
+    size: fw_header.file_size
+    type: sections
+
+types:
+  fw_header:
+    seq:
+      - id: md5sum
+        size: 0x10
+        doc: MD5 sum of whole "sections" block
+
+      - id: file_size
+        type: u4
+
+      - id: magic
+        type: u4
+        enum: magic
+
+  sections:
+    seq:
+      - id: sections
+        type: section
+        repeat: eos
+
+  section:
+    seq:
+      - id: head
+        type: head
+
+      - id: data
+        size: head.data_size
+        type:
+          switch-on: head.name.content
+          cases:
+            '"REBOOT"': reboot_data
+            '"info"': info_data
+            '"HMAC"': hmac_data
+            '"WGPKG"': wgpkg_data
+
+  name:
+    seq:
+      - id: type2_magic
+        type: u1
+        if: is_type2
+
+      - id: content
+        type: strz
+        size: 'is_type2 ? 0xf : 0x10'
+        encoding: ASCII
+
+    instances:
+      type2:
+        pos: 0
+        type: u1
+
+      is_type2:
+        value: type2 == 0x7f
```

Comment:
@ret5et The file you linked in the PR description:

> Fireware 12.5.4 Sysa-dl for OS updates from the Web UI
> example fw https://cdn.watchguard.com/SoftwareCenter/Files/XTM/12_5_4/firebox_M440_12_5_4.zip

doesn't have any section with `type2`, do you know a file that has it?

Reply:
Unfortunately, I only saw and worked with files of the first type.

Reply:
> Unfortunately, I only saw and worked with files of the first type.

OK, and how did you know the structure looks like this for `type2` if you haven't seen any test file with it? Did you use any reference where it was described?

Yeah, in general, if your work is not only based on reverse engineering but also on some existing sources, it's a very good idea to refer to them - the [`doc-ref`](https://doc.kaitai.io/user_guide.html#doc-ref) key is intended for that.

Reply:
I did a reverse engineering of some parts of the firmware, namely the part that is responsible for parsing update files. There it was seen that there are several types of files. I had no documentation. This is a small project of several evenings.